lock-api/src/main/java/com/palantir/lock/v2/TimestampRangeAndPartition.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.lock.v2;
+
+import java.util.stream.LongStream;
+
+import org.immutables.value.Value;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.palantir.lock.client.SharedConstants;
+import com.palantir.timestamp.TimestampRange;
+
+@Value.Immutable
+@JsonSerialize(as = ImmutableTimestampRangeAndPartition.class)
+@JsonDeserialize(as = ImmutableTimestampRangeAndPartition.class)
+public interface TimestampRangeAndPartition {
+    @Value.Parameter
+    TimestampRange range();
+
+    @Value.Parameter
+    int partition();
+
+    @JsonIgnore
+    default long[] getStartTimestamps() {
```

Comment:
If I asked for 10k transactions with a partition of 0, it seems weird that i'd be iterating through 160k things, rather than just 10k. For your current code, you could e.g. have a range between 0 and (upper - lower)/num partitions and then add lower and multiply by num partitions for the result...

but I'd probably just return a LongStream and compute it by being like

```
return LongStream.range(0, count).map(item -> base + item * interval);
```

Reply:
doh, missed this - yeah probably best to directly calculate the usable timestamps